pkg/controller/master/node/promote_controller.go

```diff
@@ -246,12 +248,21 @@ func (h *PromoteHandler) setPromoteResult(job *batchv1.Job, node *corev1.Node, s
 // If the cluster doesn't need to be promoted, return nil
 func selectPromoteNode(nodeList []*corev1.Node) *corev1.Node {
 	var (
-		managementNumber     int
-		managementRoleNumber int
-		canPromoteNumber     int
-		promoteNode          *corev1.Node
+		managementNumber           int
+		managementRoleNumber       int
+		canPromoteNumber           int
+		promoteNode                *corev1.Node
+		masterNodeTopologyZoneList []string
```

Comment:
shouldn't it be a set instead?

https://pkg.go.dev/k8s.io/apimachinery@v0.21.5/pkg/util/sets#String

Reply:
@weihanglo what would be the benefit of using a set?

Reply:
Never mind. I assume the list won't be too large. A slice is sufficient.